[BUG]: multiple definition of util function when use it from class method

### What happened?

When use `utils/utils.h` methods inside custom class it causes `multiple definition of` problem.

Similar problem https://stackoverflow.com/questions/6080853/c-multiple-definition-error-for-global-functions-in-the-header-file

### Version

4.0.0

### What environment are you seeing the problem in?

Platform IO, Arduino

### Board type

_No response_

### Relevant log output

```shell
MenuScreen.cpp.o (symbol from plugin): In function `MenuScreen::initialize(LcdMenu*, DisplayInterface*)':
(.text+0x0): multiple definition of `substring(char const*, unsigned char, unsigned char, char*)'
.pio\build\uno\src\main.cpp.o (symbol from plugin):(.text+0x0): first defined here
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\uno\firmware.elf] Error 1
```

--- a/src/utils/utils.h
+++ b/src/utils/utils.h
@@ -4,12 +4,12 @@
 
 #include "constants.h"
 
-void substring(const char* str, uint8_t start, uint8_t size, char* substr) {
+inline void substring(const char* str, uint8_t start, uint8_t size, char* substr) {
     strncpy(substr, str + start, size);
     substr[size] = '\0';
 }
 
-void concat(const char* first, char second, const char* third, char* result) {
+inline void concat(const char* first, char second, const char* third, char* result) {
     size_t len1 = strlen(first);
     size_t len3 = strlen(third);
 
@@ -19,19 +19,19 @@ void concat(const char* first, char second, const char* third, char* result) {
     result[len1 + 1 + len3] = '\0';
 }
 
-void concat(const char* first, char second, char* result) {
+inline void concat(const char* first, char second, char* result) {
     size_t len1 = strlen(first);
     memcpy(result, first, len1);
     result[len1] = second;
     result[len1 + 1] = '\0';
 }
 
-void concat(const char* first, const char* second, char* result) {
+inline void concat(const char* first, const char* second, char* result) {
     strcpy(result, first);
     strcat(result, second);
 }
 
-void remove(char* str, uint8_t index, uint8_t count) {
+inline void remove(char* str, uint8_t index, uint8_t count) {
     uint8_t len = strlen(str);
     if (index + count > len) {
         count = len - index;
@@ -39,26 +39,26 @@ void remove(char* str, uint8_t index, uint8_t count) {
     memmove(str + index, str + index + count, len - count - index + 1);
 }
 
-long mapProgress(uint16_t progress, long minValue, long maxValue) {
+inline long mapProgress(uint16_t progress, long minValue, long maxValue) {
     // Map the progress value to a new range (minValue to maxValue)
     return map(progress, MIN_PROGRESS, MAX_PROGRESS, minValue, maxValue);
 }
 
-float mapProgress(uint16_t progress, float minValue, float maxValue) {
+inline float mapProgress(uint16_t progress, float minValue, float maxValue) {
     // Normalize the progress value and map it to the specified floating-point
     // range
     return static_cast<float>(progress) / MAX_PROGRESS * (maxValue - minValue) +
            minValue;
 }
 
-void printCmd(const __FlashStringHelper* command) {
+inline void printCmd(const __FlashStringHelper* command) {
 #ifdef DEBUG
     Serial.print(F("#CMD# "));
     Serial.println(command);
 #endif
 }
 
-void printCmd(const __FlashStringHelper* command, const char value) {
+inline void printCmd(const __FlashStringHelper* command, const char value) {
 #ifdef DEBUG
     Serial.print(F("#CMD# "));
     Serial.print(command);
@@ -67,7 +67,7 @@ void printCmd(const __FlashStringHelper* command, const char value) {
 #endif
 }
 
-void printCmd(const __FlashStringHelper* command, const uint8_t value) {
+inline void printCmd(const __FlashStringHelper* command, const uint8_t value) {
 #ifdef DEBUG
     Serial.print(F("#CMD# "));
     Serial.print(command);
@@ -76,7 +76,7 @@ void printCmd(const __FlashStringHelper* command, const uint8_t value) {
 #endif
 }
 
-void printCmd(const __FlashStringHelper* command, const char* value) {
+inline void printCmd(const __FlashStringHelper* command, const char* value) {
 #ifdef DEBUG
     Serial.print(F("#CMD# "));
     Serial.print(command);

